src/main/java/com/gamegoo/service/manner/MannerService.java

```diff
@@ -0,0 +1,159 @@
+import com.gamegoo.repository.manner.MannerKeywordRepository;
+import com.gamegoo.repository.manner.MannerRatingKeywordRepository;
+import com.gamegoo.repository.manner.MannerRatingRepository;
+import com.gamegoo.repository.member.MemberRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class MannerService {
+    private final MemberRepository memberRepository;
+    private final MannerRatingRepository mannerRatingRepository;
+    private final MannerRatingKeywordRepository mannerRatingKeywordRepository;
+    private final MannerKeywordRepository mannerKeywordRepository;
+
+    public MannerRating insertManner(MannerRequest.mannerInsertDTO request, Long memberId){
+
+        Member member = memberRepository.findById(memberId).orElseThrow(()->new MemberHandler(ErrorStatus.MEMBER_NOT_FOUND));
+
+        // 매너평가를 받는 회원 존재 여부 검증.
+        Member targetMember = memberRepository.findById(request.getToMemberId()).orElseThrow(()->new MemberHandler(ErrorStatus.MANNER_TARGET_MEMBER_NOT_FOUND));
+
+        // 매너평가를 받는 회원 탈퇴 여부 검증.
+        if (targetMember.getBlind()){
+            throw new MemberHandler(ErrorStatus.USER_DEACTIVATED);
+        }
+
+        // 매너평가 최초 시도 여부 검증.
+        List<MannerRating> mannerRatings = mannerRatingRepository.findByToMemberId(targetMember.getId());
+        if (!mannerRatings.isEmpty()){
+            for (MannerRating mannerRating : mannerRatings) {
+                if (mannerRating.getFromMember().getId().equals(member.getId())){
+                    List<MannerRatingKeyword> mannerKeywords = mannerRatingKeywordRepository.findByMannerRating(mannerRating);
+                    if (!mannerKeywords.isEmpty()){
+                        MannerRatingKeyword mannerKeyword = mannerKeywords.get(0);  // 첫번째 키워드만 조회.
+                        if (mannerKeyword.getMannerKeyword().getIsPositive()){
+                            throw new MannerHandler(ErrorStatus.MANNER_CONFLICT);
+                        }
+                    }
+                }
+            }
+        }
+
+        // mannerKeyword 의 실제 존재 여부 검증.
+        List<MannerKeyword> mannerRatingKeywordList = new ArrayList<>();
+        request.getMannerRatingKeywordList()
+                .forEach(mannerKeywordId -> {
+                    MannerKeyword mannerKeyword = mannerKeywordRepository.findById(mannerKeywordId).orElseThrow(() -> new TempHandler(ErrorStatus._BAD_REQUEST));
+                    mannerRatingKeywordList.add(mannerKeyword);
+                });
+
+        // mannerKeyword 유형 검증.
+        for (Long keyword : request.getMannerRatingKeywordList()){
+            if (keyword < 1 || keyword > 6) {
+                throw new MannerHandler(ErrorStatus.MANNER_KEYWORD_TYPE_INVALID);
+            }
+        }
```

Comment:
이 부분도 그냥 forEach 루프 한번에 할 수 있을 것 같아요!
```suggestion
        List<MannerKeyword> mannerRatingKeywordList = new ArrayList<>();
        request.getMannerRatingKeywordList()
            .forEach(mannerKeywordId -> {
                MannerKeyword mannerKeyword = mannerKeywordRepository.findById(mannerKeywordId)
                    .orElseThrow(() -> new TempHandler(ErrorStatus.MANNER_KEYWORD_NOT_EXIST));
                if(!mannerKeyword.getIsPositive()){
                    throw new MannerHandler(ErrorStatus.MANNER_KEYWORD_TYPE_INVALID)
                }
                mannerRatingKeywordList.add(mannerKeyword);
            });
```
그리고 mannerKeyword를 찾을 수 없을 경우에도 BAD_REQUEST 말고 커스텀 에러 하나 생성해서 에러 발생시켜주는게 좋을 것 같습니다!

Reply:
[코드 리팩토링 완료]